src/vcpkg/base/checks.cpp

```diff
@@ -67,6 +70,11 @@ namespace vcpkg
     [[noreturn]] void Checks::exit_with_code(const LineInfo& line_info, const int exit_code)
     {
         Debug::println(locale_invariant_lineinfo(line_info));
+        // Collect exit code and success telemetry
+        Path file_info{line_info.file_name};
+        get_global_metrics_collector().track_bool(BoolMetric::ExitSuccess, exit_code == EXIT_SUCCESS);
+        get_global_metrics_collector().track_string(
+            StringMetric::ExitCode, fmt::format("{};{};{}", exit_code, file_info.filename(), line_info.line_number));
```

Comment:
filename could include PII (if the user built the tool themselves), so we should instead use the function name. We just need to add `__func__` to the LineInfo.

Reply:
Yeah. I assumed that stripping the parent path would be enough but you're right, highly unlikely but possible. I'll add __func__ to LineInfo